### PR TITLE
hugo: Allow preselected tags on search

### PR DESCRIPTION
**Pull request by Wies-UsMedia**
See commit message.

To test, go to https://deploy-preview-477--cue.netlify.app/examples/shortcodes/search/ and you should be able to add new tags, as well as remove the preselected tags.

### CODE REVIEW

**Review by JorindeUsMedia — APPROVED** [review]

File: hugo/assets/ts/widgets/search-filter.ts [hunk 1]
```diff
@@ -139,21 +140,40 @@ export class SearchFilter extends BaseWidget {
...
 
-        if (!query) {
-            this.selectedItems = [];
-        } else {
-            this.selectedItems = this.parsedQuery.facets[this.filterName] ?? [];
+        let urlParamsChanged = false;
+        let selectedItems = this.parsedQuery.facets[this.filterName] ?? [];
+
+        // Get preselected items if no items set in url (if set in urL: url wins from preselected)
+        if (selectedItems.length === 0 && this.element.dataset.preselectedItems && this.element.dataset.preselectedItems !== '') {
```
Comment (JorindeUsMedia) [inline hugo/assets/ts/widgets/search-filter.ts:164]: As discussed in our call: we should only add preselected tags  when there is no query at all and not just when there is no tags in the query. So change it to if !(query) I guess or !this.parsedQuery ( not sure what the value is when there is no query)

**Review by JorindeUsMedia — CHANGES_REQUESTED** [review]

File: hugo/assets/ts/widgets/search-filter.ts (hunk 1 above)
Comment (Wies-UsMedia) [reply hugo/assets/ts/widgets/search-filter.ts]: Should be updated now.

**Comment by jpluscplusm** [comment]
@Wies-UsMedia Is this ready for (re-)review?

**Comment by Wies-UsMedia** [comment]
> @Wies-UsMedia Is this ready for (re-)review?

It is yep!

**Comment by jpluscplusm** [comment]
CI passed. Preview is here: https://cl-1203609-1--cue-cls.netlify.app/

**Review by jpluscplusm — APPROVED** [review]
Reviewed via https://cl-1203609-1--cue-cls.netlify.app/examples/shortcodes/search/.

We still have a corner-corner case where removing the last tag leaves an empty `q` query string param, but then resets the tags to the predefined set, but that's sufficiently niche that I'll just track it as a new issue that we can address if it becomes a problem.

**Pull request closed without merging**